kratos/tests/strategies/builder_and_solvers/test_builder_and_solver.cpp

```diff
@@ -0,0 +1,841 @@
+            KRATOS_CHECK_LESS_EQUAL(std::abs((rA(6,6) - 2.069e+09)/rA(6,6)), tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs((rA(7,7) - 1)/rA(7,7)), tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs((rA(8,8) - 1)/rA(8,8)), tolerance);
+        }
+
+        /**
+         * Checks if the block builder and solver with constraints performs correctly the resolution of the system
+         */
+        KRATOS_TEST_CASE_IN_SUITE(BasicDisplacementBlockBuilderAndSolverWithConstraints, KratosCoreFastSuite)
+        {
+            if (!KratosComponents<Element>::Has("TrussElement3D2N")) {
+                std::cout << "Please compile the StructuralMechanicsApplication in order to run this test" << std::endl;
+                return void();
+            }
+
+            Model current_model;
+            ModelPart& r_model_part = current_model.CreateModelPart("Main", 3);
+
+            BasicTestBuilderAndSolverDisplacement(r_model_part, true);
+
+            SchemeType::Pointer p_scheme = SchemeType::Pointer( new ResidualBasedIncrementalUpdateStaticSchemeType() );
+            LinearSolverType::Pointer p_solver = LinearSolverType::Pointer( new SkylineLUFactorizationSolverType() );
+            BuilderAndSolverType::Pointer p_builder_and_solver = BuilderAndSolverType::Pointer( new ResidualBasedBlockBuilderAndSolverWithConstraintsType(p_solver) );
+
+            const SparseSpaceType::MatrixType& rA = BuildSystem(r_model_part, p_scheme, p_builder_and_solver);
+
+            // To create the solution of reference
+//             DebugLHS(rA);
+
+            // The solution check
+            constexpr double tolerance = 1e-4;
+            KRATOS_CHECK(rA.size1() == 9);
+            KRATOS_CHECK(rA.size2() == 9);
+            KRATOS_CHECK_LESS_EQUAL(std::abs((rA(0,0) - 2.069e+09)/rA(0,0)), tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs((rA(1,1) - 1)/rA(1,1)), tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs((rA(2,2) - 1)/rA(2,2)), tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs((rA(3,3) - 2.069e+09)/rA(3,3)), tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs((rA(4,4) - 1)/rA(4,4)), tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs((rA(5,5) - 1)/rA(5,5)), tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs((rA(6,6) - 2.069e+09)/rA(6,6)), tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs((rA(7,7) - 1)/rA(7,7)), tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs((rA(8,8) - 1)/rA(8,8)), tolerance);
+        }
+
+        /**
+         * Checks if the elimination builder and solver performs correctly the resolution of the system
+         */
+        KRATOS_TEST_CASE_IN_SUITE(BasicDisplacementEliminationBuilderAndSolver, KratosCoreFastSuite)
+        {
+            if (!KratosComponents<Element>::Has("TrussElement3D2N")) {
+                std::cout << "Please compile the StructuralMechanicsApplication in order to run this test" << std::endl;
```

Comment:
sholdn't this be a warning? also, is this the "official" way of skipping a test?

Reply:
This is the way I has been using in my tests for mixed applications. And yes it works without compiling the Structural